addons/custom-block-shape/userscript.js

```diff
@@ -0,0 +1,246 @@
+export default async function ({ addon, global, console }) {
+  var multiplier = addon.settings.get("paddingSize") / 100;
+  var BlocklyInstance = await addon.tab.traps.getBlockly();
+
+  (function (Blockly) {
+    const BlockSvg = BlocklyInstance.BlockSvg;
+    var vm = addon.tab.traps.vm;
+
+    const { GRID_UNIT } = BlockSvg;
+    var cornerSize = 1;
+    var notchSize = 1;
+
+    function updateAllBlocks() {
+      if (vm.editingTarget) {
+        vm.emitWorkspaceUpdate();
+        Blockly.getMainWorkspace().getToolbox().flyout_.workspace_.clear();
+        Blockly.getMainWorkspace().getToolbox().refreshSelection();
+      }
+    }
```

Comment:
So apparently this still causes a bunch issues of varying severity, my bad.

I think they're all fixed here, for real this time:

```suggestion
    function updateAllBlocks() {
      const workspace = Blockly.getMainWorkspace();
      if (workspace) {
        if (vm.editingTarget) {
          vm.emitWorkspaceUpdate();
        }
        const flyout = workspace.getFlyout();
        if (flyout) {
          const flyoutWorkspace = flyout.getWorkspace();
          Blockly.Xml.clearWorkspaceAndLoadFromXml(Blockly.Xml.workspaceToDom(flyoutWorkspace), flyoutWorkspace);
          workspace.getToolbox().refreshSelection();
          workspace.toolboxRefreshEnabled_ = true;
        }
      }
    }
```